scripts/issue/354/upgrade_remaining_setts_pause.py

```diff
@@ -0,0 +1,259 @@
+                    console.print(f"[green]Queueing upgrade on timelock for {key} to SettV1_1h.sol[/green]")
+                    safe.badger.queue_timelock(
+                        target_addr=DEV_PROXY,
+                        signature="upgrade(address,address)",
+                        data=encode_abi(
+                            ["address", "address"],
+                            [address, settV1_1h_logic],
+                        ),
+                        dump_dir="data/badger/timelock/upgrade_remaining_setts/",
+                        delay_in_days=2.3,
+                    )
+                else:
+                    console.print(f"[green]{key} has already been upgraded[/green]")
+
+            elif key in SETTV4_KEYS:
+                if get_implementation(address) != settV4h_logic:
+                    console.print(f"[green]Queueing upgrade on timelock for {key} to SettV4h.sol[/green]")
+                    safe.badger.queue_timelock(
+                        target_addr=DEV_PROXY,
+                        signature="upgrade(address,address)",
+                        data=encode_abi(
+                            ["address", "address"],
+                            [address, settV4h_logic],
+                        ),
+                        dump_dir="data/badger/timelock/upgrade_remaining_setts/",
+                        delay_in_days=2.3,
+                    )
+                else:
+                    console.print(f"[green]{key} has already been upgraded[/green]")
+
+            elif key in SPECIAL_SETTS:
+                console.print(f"[red]No new logic deployed for {key} yet[/red]")
+
+            else:
+                console.print(f"[red]{key} is not categorized![/red]")
+
+        safe.post_safe_tx()
+
+    else:
+        for key in SETTV1_KEYS+SETTV1_1_KEYS+SETTV4_KEYS:
+            execute_timelock(
+                safe.badger.timelock, 
+                "data/badger/timelock/upgrade_remaining_setts/", 
+                key,
+                simulation,
+            )
+
+        # Executing all upgrades together
+        safe.post_safe_tx()
+
+
+def execute_timelock(timelock, queueTx_dir, key, simulation):
+    if simulation == 'false':
+        console.print(f"Processing upgrade and patch for {key}...")
+        path = os.path.dirname(queueTx_dir)
+        directory = os.fsencode(path)
+
+        sett = interface.ISettV4h(vaults[key])
+
+        for file in os.listdir(directory):
+            filename = os.fsdecode(file)
+            if not filename.endswith('.json'):
+                continue
+            txHash = filename.replace(".json", "")
+
+            with open(f"{queueTx_dir}{filename}") as f:
+                    tx = json.load(f)
+
+            # Get Sett address from upgrade Tx
+            target_sett = web3.toChecksumAddress('0x' + tx['data'][24:64])
+
+            if sett.address == target_sett:
+                if timelock.queuedTransactions(txHash) == True:
```

Comment:
i think in this section we could rely similarly on this [method](https://github.com/Badger-Finance/badger-multisig/blob/main/great_ape_safe/ape_api/badger.py#L196) in the badger class

Reply:
@petrovska-petro I think we need this implementation of the function so we can sanity check the variables before and after the upgrade

Reply:
i guess you could do it with the `safe.badger.execute_timelock` if you would dump jsons to subdirs per sett maybe?

anyway end result will still be the same, i see no need to refactor.

Reply:
yep, will result on the same agree even if refactor 👍